### PR TITLE
[P4-2005] Support concept of "Add another" from the framework

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

Added a major new feature to the Person Escort Record area of the application. This adds support for capturing Property Information using a new "Add another" pattern.

### Why did it change

Property is the last major section of the Person Escort Record that needs to be digitally captured. This supports the changed to the form wizard and framework needed in order to support adding multiple items from the framework.

### Issue tracking

- [P4-2005](https://dsdmoj.atlassian.net/browse/P4-2005)

## Screenshots

<kbd></kbd>

<kbd></kbd>

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: app/person-escort-record/controllers/framework-step.js [hunk 1]
```diff
@@ -80,12 +81,12 @@ class FrameworkStepController extends FormWizardController {
...
 
   getValues(req, res, callback) {
-    const { errors, errorValues } = req.sessionModel.toJSON()
+    const { errors, errorValues, ...values } = req.sessionModel.toJSON()
     const fields = req.form.options.fields
     const responses = req.personEscortRecord.responses
     const savedValues = responses
       .filter(response => fields[response.question?.key])
-      .filter(response => response.value)
+      // .filter(response => response.value)
```
Comment (solidgoldpig) [inline app/person-escort-record/controllers/framework-step.js:89]: ??

**Review by JadeNConroy — APPROVED** [review]

File: common/controllers/form-wizard.js [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
 class FormController extends Controller {
   middlewareSetup() {
     super.middlewareSetup()
+    this.use(this.setupAddMultipleFields)
     this.use(this.setupConditionalFields)
     this.use(this.setFieldContext)
   }
 
+  middlewareActions() {
+    super.middlewareActions()
```
Comment (solidgoldpig) [inline common/controllers/form-wizard.js:16]: What's the point of just calling the `super` method?

Comment (teneightfive) [reply common/controllers/form-wizard.js:16]: None, I forgot to remove this after changing the above...

File: common/controllers/form-wizard.js [hunk 3]
```diff
@@ -30,6 +51,59 @@ class FormController extends Controller {
     next()
   }
 
+  post(req, res, next) {
+    // QUESTION: Not sure if this should live in `post`/`process`/somewhere else
```
Comment (solidgoldpig) [inline common/controllers/form-wizard.js:55]: This seems like the best place to me - if you put anywhere later in the lifecycle such as `process` then it's going to get messy if another controller needs to override the method[0] - and the later it is, the more likely that's going to be.

[0] Unless the override stuff doesn't need to run until __after__ the existing super stuff

File: app/person-escort-record/controllers/framework-step.js [hunk 4]
```diff
@@ -32,7 +32,14 @@ class FrameworkStepController extends FormWizardController {
 
   middlewareSetup() {
     super.middlewareSetup()
+    // TODO: Find a better way to fix the ordering of these middleware
```
Comment (solidgoldpig) [inline app/person-escort-record/controllers/framework-step.js:35]: Maybe I'm getting the wrong end of the stick, but could you add a noop `setInitialValues` to the form-wizard controller and simply override that here?

**Review by solidgoldpig — APPROVED** [review]

File: app/person-escort-record/controllers/framework-step.js (hunk 4 above)
Comment (teneightfive) [reply app/person-escort-record/controllers/framework-step.js:35]: Ooo, that's a very good idea :)

**Pull request merged**